src/components/data-management/SpeciesSelector.jsx

```diff
@@ -3,20 +3,38 @@ import PropTypes from 'prop-types';
 import {
   Select, Typography, Space, Divider, Skeleton,
 } from 'antd';
-import _ from 'lodash';
 
 const { Text } = Typography;
 
 const SpeciesSelector = (props) => {
-  const { data, onChange } = props;
+  const { data, value, onChange } = props;
 
-  if (!data) {
+  if (!data || data.length === 0) {
     return <Skeleton.Input style={{ width: 300 }} size='small' />;
   }
 
+  const createLabel = (scientificName, displayName) => (
+    <Space direction='vertical'>
+      <Text type='primary'>{displayName}</Text>
+      <Text type='secondary'>
+        {scientificName}
+      </Text>
+    </Space>
+  );
+
+  const labeledValue = (organismId) => {
+    const organism = data.find((entry) => entry.id === organismId);
+
+    return {
+      value: organismId,
+      label: !organismId ? <></> : createLabel(organism.scientific_name, organism.display_name),
+    };
+  };
+
   return (
     <Select
-      onChange={(value, option) => onChange(option.displayName)}
+      value={labeledValue(value)}
```

Comment:
I can definitely see that the need for this usage of `createLabel` and `labeledValue` is coming from this line and also from line 75 below. I would definitely consider splitting these things up and making them less coupled. Maybe the original way of how label was defined on lines 57-64 that are deleted was better and all you need is a small function called "getLabelName" for line 36, without the return statement of `label: !organismId ? <></> : createLabel(organism.scientific_name, organism.display_name),` is all you need

Reply:
Cleaned up by removing `labelInValue`, which requires the value to be of object `{ value: string, label : ReactNode}`, so now we can just pass `value` instead of having to construct the entire React node.